kong/plugins/apifortress/capturer.lua

```diff
@@ -0,0 +1,31 @@
+-- ©2016 API Fortress Inc.
+-- Captures the response payloads as they go through the wire
+local _M = {}
+
+local function read_response_body()
+	local chunk, eof = ngx.arg[1], ngx.arg[2]
+	local buffered = ngx.ctx.buffered
+	if not buffered then
+		buffered = {}
+		ngx.ctx.buffered = buffered
```

Comment:
Oops, my bad. I meant it the other way around: why not using `captured_body` as the buffer itself instead of needing another buffer? `ngx.ctx` calls are expensive as they required metatable lookups. The less the better! :)

See mashape-analytics's `handler.lua` for body buffering.